### DISCUSSION TITLE
Greenness Index 7 day forecast chart

### DISCUSSION THREAD

**Discussion by myromeo** [BottlecapDave/HomeAssistant-OctopusEnergy#760] [Show and tell]
Just wanted to share an Apex Chart I've created that uses a data_generator to extract and plot the next 7 days greenness index on a simple graph, maybe @BottlecapDave this could be included in the documentation? 

Just need to enable the entity and update the series/entity row to suit your instance. 

YAML here: https://pastebin.com/G2MqD1U6

Screenshot here:

<img width="506" alt="Screenshot 2024-02-15 at 21 28 14" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/60740643/f6f9a6f8-2570-4ecf-8f2f-14ed42c877cc">

**Comment by BottlecapDave** [comment]
Feel free to raise a merge request with the addition in the community section.

**Comment by xlibbyx** [comment]
In case it's useful to anyone, I knocked up a template sensor to show the values on my dashboard. Change the name and id, and iterate the '[0]' for subsequent days.

```
- sensor:
  - name: Green Today
    unique_id: green_today
    state: >
      {% set greenness_list = state_attr('sensor.octopus_energy_<your account number here>_greenness_forecast_current_index','forecast') %}
      {% set greenness = greenness_list[0]["greenness_score"] %}
      {{ greenness }}
    icon: >-
      {% set value = states('sensor.green_today') | float(none) %}
      {% if value > 60 %}
        mdi:leaf
      {% else %}
        mdi:leaf-off
      {% endif %}
```

**Reply by BottlecapDave** [reply]
Be careful. The greenness score is only valid between the dates specified, as OE are currently only tracking wind. In the future, there should be bigger forecasts as they track other measurements (e.g. solar). Therefore with this sensor, it will be true throughout most of the day before which could be invalid.

**Comment by thewishy** [comment]
I made a minor mod to list day-names rather than dates, I find that works easier for me. Obviously myromeo still deserves all the credit

https://pastebin.com/EsbAk2ip

The logic is slightly changed vs the main greener days website, they prefer "Plug in on Tuesday for it to charge on Wednesday morning", whereas this is "It's greenest to charge on Wednesday morning".

**Reply by myromeo** [reply]
Nice one!
